File: leetcode_ly/c++/Bit/SingleNumberTwo.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        vector<int> ans(32);
        int res = 0;
        for(int i = 0; i < 32; i++){
            for(int j = 0; j < nums.size(); j++){
                ans[i] += (nums[j] >> i) & 1;
            }
            res |= (ans[i] % 3) << i;
        }
        return res;
    }
};

File: leetcode_ly/java/Bit Manipulation && Design/SingleNumberTwo.java
public class Solution {
    public int singleNumber(int[] nums) {
        if(nums == null || nums.length == 0) return 0;
        int res = 0;
        for(int i = 0; i < 32; i++){
            int sum = 0;
            for(int j = 0; j < nums.length; j++){
                if(((nums[j] >>> i) & 1) == 1){
                    sum++;
                    sum = sum % 3;
                }
            }
            if(sum != 0){
                res |= (sum << i);
            }
        }
        return res;
    }
}